### ISSUE TITLE
[QUESTION]- Suggestion on using images in apps/games

### ISSUE THREAD

**Issue by Gokulv617** [Gamua/Adobe-Runtime-Support#578]
**AIR - 33.1.1.345**

I have a game in playstore for a quite long time and its **ANR rate was around 0.5% to 0.6%**. The main ANR was “**Input dispatching timed out (Waiting to send non-key event because the touched window has not finished processing certain input events that were delivered to it over 500.0ms ago. Wait queue length: 6. Wait queue head age: 13616.5ms.)**”. Recently 20 days before I updated my game adding Christmas Theme(changing background images) and the ANR rises. **Now the ANR is 0.86%**. I use .png format images generated from photoshop. 

I have 3 questions now, 
•	Please let me know whether the images affect the performance and causes ANR ? what might be the reason ?
•	What type of image format is more efficient in rendering process .png/.jpeg/vector or any other ?
•	Also please give me your suggestion on how to use images for smooth rendering in AIR ?

I referred this - (https://github.com/Gamua/Adobe-Runtime-Support/issues/448)

**Comment by berkayk** [comment]
I am also having the same problem.

**Comment by Gokulv617** [comment]
ANIMATE CC 21.0.1
AIR SDK – 33.1.1.345
 
I’m concentrating on rendering process. Reduced all the resolution of images(.png) I used in this project to the exact sizes it should be. The stage size I use is 800x480. I'm checking whether this improves the performance. Can someone suggest the type of "render mode" I should use for better performance ?

ANIMATE CC provides 4 different render mode's
- AUTO
- CPU
- GPU
- DIRECT
 
I experimented with "GPU" and it is making my game slower in low end devices. Used "AUTO", I'm not able to see any differences from "CPU" to "AUTO" mode. Both "CPU" and "AUTO" gives me a same kind of performance.

**Note: “Render mode” I used for the project is “CPU”.**
